x-pack/plugins/security_solution/server/lib/detection_engine/rules/saved_object_mappings.ts

```diff
@@ -47,11 +53,30 @@ export const ruleStatusSavedObjectMappings: SavedObjectsType['mappings'] = {
   },
 };
 
+const truncateMessageFields = (
```

Comment:
for backwards compatibility we had to make the types for migrations very lax by default, but since security_solution usually has very good types for your data could you use the `SavedObjectMigrationFn` type to provide more type-safety?

https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/migrations/types.ts#L12-L48